src/fides/api/task/conditional_dependencies/privacy_request/privacy_request_data.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+from typing import Any, Optional
+from uuid import UUID
+
+from pydantic import BaseModel
+from pydantic.v1.utils import deep_update
+
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest
+from fides.api.schemas.policy import Policy as PolicySchema
+from fides.api.task.conditional_dependencies.privacy_request.convenience_fields import (
+    get_policy_convenience_fields,
+)
+from fides.api.task.conditional_dependencies.util import (
+    extract_nested_field_value,
+    set_nested_value,
+)
+
+
+class PrivacyRequestDataTransformer:
+    """
+    Transforms a PrivacyRequest ORM object into a dictionary structure for condition evaluation.
+
+    This class extracts the fields referenced in conditions and builds a minimal nested structure
+    for evaluation. Privacy request fields must be prefixed with the "privacy_request" namespace.
+
+    Note: Privacy request relationships are eager loaded in the execute_request_tasks.py file.
+    """
+
+    def __init__(self, privacy_request: PrivacyRequest):
+        self.privacy_request = privacy_request
+        self.policy_data = self._transform_policy_object(privacy_request.policy)
+        self.identity_data = self.privacy_request.get_persisted_identity()
+        self.custom_privacy_request_fields_data = (
+            self.privacy_request.get_persisted_custom_privacy_request_fields()
+        )
+
+    def to_evaluation_data(self, field_addresses: set[str]) -> dict[str, Any]:
+        """
+        Transforms the PrivacyRequest into an evaluation-ready dictionary structure.
```

Comment:
If we return Any, how does the evaluator know what to do with the values? Will it be expected to type check everything or do we need to transform it into something a bit more consistent that has a predictable interface?